### PR TITLE
Add initial support for Tuya covering Ds82

### PR DESCRIPTION
**Pull request by Smanar**
- Add support for DS82 model.
- Correct the "reverse" command for all tuya covering using the tuya cluster.
- Make the code more generic.

### CODE REVIEW

**Comment by Mimiix** [comment]
Don't merge yet, I'll probably have a user that needs 1 added @Smanar

**Comment by Mimiix** [comment]
Seems that TS0601 is missing, could that be correct?

**Comment by Mimiix** [comment]
@Smanar Send you a pm discord.

**Comment by Smanar** [comment]
For information, with this code all this list can work

```
    {"_TYST11_wmcdj3aq", "mcdj3aq", "Zemismart", "Tuya_COVD ZM25TQ"},
    {"_TZE200_wmcdj3aq", "TS0601", "Zemismart", "Tuya_COVD ZM25TQ"},
    {"_TZE200_fzo2pocs", "TS0601", "Zemismart", "Tuya_COVD ZM25TQ"},
    {"_TYST11_xu1rkty3", "u1rkty3", "Smart Home", "Tuya_COVD DT82LEMA-1.2N"},
    {"_TZE200_xuzcvlku", "TS0601", "Zemismart", "Tuya_COVD M515EGB"},
    {"_TZE200_zah67ekd", "TS0601", "MoesHouse / Livolo", "Tuya_COVD AM43-0.45-40"},
    {"_TZE200_nogaemzt", "TS0601", "Tuya", "Tuya_COVD YS-MT750"},
    {"_TZE200_zpzndjez", "TS0601", "Tuya", "Tuya_COVD DS82"},
    {"_TZE200_cowvfni3", "TS0601", "Zemismart", "Tuya_COVD ZM79E-DT"},
    {"_TZE200_5zbp6j0u", "TS0601", "Tuya/Zemismart", "Tuya_COVD DT82LEMA-1.2N"},
    {"_TZE200_fdtjuw7u", "TS0601", "Yushun", "Tuya_COVD YS-MT750"},
    {"_TZE200_bqcqqjpb", "TS0601", "Yushun", "Tuya_COVD YS-MT750"},
```

**Comment by micke** [comment]
I've built a docker container with this PR's changes, available at `lisinge/deconz:ds82` if anyone wants to try it out.

**Comment by Kllrv** [comment]
> lisinge/deconz:ds82

Helped me test more. Can now confirm the reverse command doesn't change actual movement direction of open/close, so close command may actually open the curtains.

Apart from that, it displays properly as a cover device in Home Assistant and this means a bubble gum fix is an option (but not prefered).

**Comment by Smanar** [comment]
Yep without this PR the reverse command not working at all, you can set if true or false, the api send the same command.

File: rest_lights.cpp [hunk 1]
```diff
@@ -2381,10 +2381,10 @@ int DeRestPluginPrivate::setLightAttributes(const ApiRequest &req, ApiResponse &
         //taskRef.transitionTime = 4;
         //taskRef.onTime = 0;
 
-        QByteArray direction = QByteArray("\x01\x00", 2);
+        QByteArray direction = QByteArray("\x00\x00", 2);
```
Comment (imammedo) [inline rest_lights.cpp:2384]: I really would like DS82 support to be merged, so do not consider it as a critique.

commit subject and commit  message should really say what patch is doing and why it's changing theses lines.
That will help maintainer to understand its purpose and why it's needed.

Comment (Smanar) [reply rest_lights.cpp:2384]: Is written on the message ?

>Correct the "reverse" command for all tuya covering using the tuya cluster.

Comment (imammedo) [reply rest_lights.cpp:2384]: yes, something along these lines
subject could be short like you suggest

commit message could be multiline a provide more verbose description what commit does an why it does something. 
the same applies to other commits.

it helps not only maintainer but also documents why/what change does, so others (including yourself in the future when you forget about it) can understand purpose of the commit.

**Comment by manup** [comment]
Pretty cool that the PR replaces some of the cryptic manufacturer name checks.

The code looks fine to me, is it ready for merging?

**Comment by Smanar** [comment]
Yep, ready to merge.

**Comment by micke** [comment]
👏 👏 👏  Great work Smanar!

**Comment by Smanar** [comment]
But it can't work correclty without this one https://github.com/dresden-elektronik/deconz-rest-plugin/pull/4616

**Pull request merged**